### PR TITLE
Add client/app fixtures and tests for login view

### PR DESCRIPTION
**Pull request by davidag**
Adds:
- client and app pytest fixtures using the new `create_app()` factory method.
- Test login credentials in login view.
- Test that session cookie is set after correct login.
- Test that login redirects to default calendar if credentials were valid.

### CODE REVIEW

**Comment by Kartones** [comment]
IIRC, I prepared the code to support that fact. Check https://github.com/Kartones/flask-calendar/blob/master/flask_calendar/actions.py#L20, if you `mock.patch` the value of `config.FAILED_LOGIN_DELAY_BASE` to be `0`, failed login attempts won't force you to wait, which I guess is what makes the tests really slow.

Let me know if that helps or if still slow where it seems to take more time and we can come up with a solution.

File: test/test_app.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ('a_username', 'wrong_password', False),
+    ('a_username', 'a_password', True)
+))
+def test_login_credentials(client: FlaskClient, username: str, password: str, success: bool) -> None:
+    rv = client.post('/do_login', data=dict(
+        username=username,
+        password=password
+    ), follow_redirects=True)
+    cookie = next((c for c in client.cookie_jar), None)
+    assert rv.status_code == 200
```
Comment (Kartones) [inline test/test_app.py]: If you want to be purist, this sounds like two tests:
- one to check responses of login attempts (`test_login_credentials` sounds good)
- another to check cookie(s) once logged in (`test_session_id_cookie_present_when_logged_in` or similar)

I don't always follow this rule so just take it as a suggestion for you to decide, not a requirement.

File: test/test_app.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+@pytest.mark.parametrize(('username', 'password', 'success'), (
+    ('', '', False),
+    ('wrong_username', '', False),
+    ('a_username', '', False),
+    ('a_username', 'wrong_password', False),
+    ('a_username', 'a_password', True)
+))
+def test_login_credentials(client: FlaskClient, username: str, password: str, success: bool) -> None:
+    rv = client.post('/do_login', data=dict(
```
Comment (Kartones) [inline test/test_app.py]: A name like `response` is way more meaningful than `rv`, don't you think?

Comment (davidag) [reply test/test_app.py]: Sure!

File: test/test_app.py (hunk 1 above)
Comment (davidag) [reply test/test_app.py]: Having two tests, one for credentials and one for the session cookie, seems the way to go. This forces me to check if the credentials were valid or not in a different way.

**Comment by davidag** [comment]
@Kartones Forgot to mention that this PR was ready for review! :man_dancing:

**Pull request merged**